nntrainer/dataset/data_producers.h

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   data_producers.h
+ * @date   09 July 2021
+ * @brief  This file contains data producer interface
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+#ifndef __DATA_PRODUCERS_H__
+#define __DATA_PRODUCERS_H__
+
+#include <functional>
+#include <limits>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <tensor.h>
+#include <tensor_dim.h>
+
+namespace nntrainer {
+
+/**
+ * @brief DataProducer interface used to abstract data provider
+ *
+ */
+class DataProducer {
+public:
+  using Iteration = std::tuple<bool, std::vector<Tensor>, std::vector<Tensor>>;
```

Comment:
please add a comment describing/naming the three type variables.

Reply:
Done